src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let current = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        current = {
+          ...current, ...action.extraData,
+        };
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete current[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        current = {};
+        break;
+      }
+
+      default: {
```

Comment:
Empty default case is unnecessary. It can be removed if it doesn't serve any purpose.

Reply:
[CODE STYLE]: switch/case should always have default case for error handling.